packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -712,6 +747,9 @@ function updateStaticChildren(c1: VNodes, c2: VNodes, parent: ParentNode) {
                 mount(n2, parent, anchor);
                 anchor = n2.elm!;
             }
+        } else if (n2 !== null) {
+            // it is hoisted
+            anchor = n2.elm!;
```

Comment:
I don't understand this part, could you explain?

Reply:
in line 736 there's a check `n2 !== n1`, which previous to this pr was only true when both nodes were `null`. now (remember, there's only 1 per module) that will be the also the case for hoisted vnodes, and we need to go via patch line 740, but there's nothing to patch, therefore we only need to update the anchor.

Reply:
Could you add a code comment? I think this whole function is very complex and could benefit from it. 🙂